gammapy/background/models.py

```diff
@@ -71,3 +82,724 @@ def evaluate(self, x, y):
+
+    def to_table(self):
+        """Convert cube background model to astropy table format.
+
+        The name of the table is stored in the table meta information
+        under the keyword 'name'.
+
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            table containing the bg cube
+        """
+        # data arrays
+        a_detx_lo = Quantity([self.detx_bins[:-1]])
+        a_detx_hi = Quantity([self.detx_bins[1:]])
+        a_dety_lo = Quantity([self.dety_bins[:-1]])
+        a_dety_hi = Quantity([self.dety_bins[1:]])
+        a_energy_lo = Quantity([self.energy_bins[:-1]])
+        a_energy_hi = Quantity([self.energy_bins[1:]])
+        a_bg = Quantity([self.background])
+
+        # table
+        table = Table()
+        table['DETX_LO'] = a_detx_lo
+        table['DETX_HI'] = a_detx_hi
+        table['DETY_LO'] = a_dety_lo
+        table['DETY_HI'] = a_dety_hi
+        table['ENERG_LO'] = a_energy_lo
+        table['ENERG_HI'] = a_energy_hi
+        table['Bgd'] = a_bg
+
+        table.meta['E_THRES'] = a_energy_lo.flatten()[0].value
+
+        # name
+        table.meta['name'] = 'BACKGROUND'
+
+        return table
+
+    def to_fits_bin_table(self):
+        """Convert cube background model to binary table fits format.
+
+        Returns
+        -------
+        tbhdu : `~astropy.io.fits.BinTableHDU`
+            table containing the bg cube
+        """
+        # build astropy table and convert it to fits bin table
+        return _table_to_fits_bin_table(self.to_table())
+
+    def to_fits_image(self):
+        """Convert cube background model to image fits format.
+
+        Returns
+        -------
+        hdu_list : `~astropy.io.fits.HDUList`
+            HDU list with:
+
+            * one `~astropy.io.fits.ImageHDU` image for the bg cube
+
+            * one `~astropy.io.fits.BinTableHDU` table for the energy binning
+        """
+        # data
+        imhdu = fits.ImageHDU(data=self.background.value, name='BACKGROUND')
+        # add some important header information
+        imhdu.header['BG_UNIT'] = '{0.unit:FITS}'.format(self.background)
+        imhdu.header['CTYPE3'] = 'ENERGY'
+        imhdu.header['CUNIT3'] = '{0.unit:FITS}'.format(self.energy_bins)
+        imhdu.header['E_THRES'] = self.energy_bins[0].value
+
+        # get WCS object and write it out as a FITS header
+        wcs_header = self.det_wcs.to_header()
+
+        # transferring header values
+        imhdu.header.update(wcs_header)
+
+        # get energy values as a table HDU, via an astropy table
+        energy_table = Table()
+        energy_table['ENERGY_BIN_EDGES'] = self.energy_bins
+        energy_table.meta['name'] = 'ENERGY_BINS'
+
+        enhdu = _table_to_fits_bin_table(energy_table)
+
+        hdu_list = fits.HDUList([fits.PrimaryHDU(), imhdu, enhdu])
+
+        return hdu_list
+
+    def write(self, outfile, format='table', write_kwargs=None):
```

Comment:
I'd prefer you use `**kwargs` here instead of `write_kwargs=None`.
That's the common pattern and it should work.

I remember this has been discussed before, and there was some issue.
But please, let's try to make this work and re-discuss if needed.
One thing you have to know is that `**kwargs` always has to be at the end.